packages/sit-onyx/src/composables/useMoreList.ts

```diff
@@ -0,0 +1,216 @@
+   */
+  disabled: Ref<boolean>;
+}>;
+
+export type UseMoreListOptions = {
+  /**
+   * Vue template ref for the parent element containing the list of components.
+   */
+  parentRef: Ref<HTMLOrInstanceRef>;
+  /**
+   * Refs for the individual components in the list.
+   */
+  componentRefs: Map<string, Ref<HTMLOrInstanceRef>>;
+  /**
+   * Whether the intersection observer should be disabled (e.g. when more feature is currently not needed due to mobile layout).
+   */
+  disabled?: Ref<boolean>;
+};
+
+/**
+ * Composable for managing a list of components where e.g. a "+3" more indicator should be shown if not all components
+ * fit into the available width.
+ *
+ * @example
+ *
+ * ```vue
+ * <script lang="ts" setup>
+ * import { ref, type ComponentInstance } from "vue";
+ * import { useMore } from "../../composables/useMore";
+ * import OnyxNavButton from "../OnyxNavBar/modules/OnyxNavButton/OnyxNavButton.vue";
+ *
+ * const parentRef = ref<HTMLElement>();
+ * const componentRefs = ref<ComponentInstance<typeof OnyxNavButton>[]>([]);
+ *
+ * const { visibleElements, hiddenElements } = useMore({ parentRef, componentRefs });
+ * </script>
+ *
+ * <template>
+ *   <div ref="parentRef" class="onyx-more">
+ *     <OnyxNavButton v-for="i in 16" ref="componentRefs" :key="i" :label="`Nav button ${i}`" />
+ *   </div>
+ * </template>
+ *
+ * <style lang="scss">
+ * @use "../../styles/mixins/layers.scss";
+ *
+ * .onyx-more {
+ *   @include layers.component() {
+ *     display: flex;
+ *     align-items: center;
+ *     overflow-x: clip;
+ *   }
+ * }
+ * </style>
+ * ```
+ */
+export const useMoreList = (options: UseMoreListOptions) => {
+  const visibleElements = ref<string[]>([]);
+
+  const hiddenElements = computed(() => {
+    return Array.from(options.componentRefs.keys()).filter(
+      (key) => !visibleElements.value.includes(key),
+    );
+  });
+
+  const observer = ref<IntersectionObserver>();
+  onBeforeUnmount(() => observer.value?.disconnect());
+
+  watch(
+    [options.parentRef, options.componentRefs, options.disabled],
+    () => {
+      observer.value?.disconnect(); // reset observer before all changes
+
+      const root = getTemplateRefElement(options.parentRef.value);
+      if (!root || options.disabled?.value) {
+        visibleElements.value = [];
+        return;
+      }
+
+      observer.value = new IntersectionObserver(
+        (changeEntries) => {
+          // changeEntries contains all changed component visibilities (not all available components)
+          // if component is shown, intersectionRatio is 1, otherwise its completely or partially hidden
+          const shownIds: string[] = [];
+          const hiddenIds: string[] = [];
+
+          changeEntries.forEach((entry) => {
+            const elementId = Array.from(options.componentRefs).find(([_, element]) => {
+              return getTemplateRefElement(unref(element))?.isSameNode(entry.target);
+            })?.[0];
```

Comment:
`isSameNode` is a [legacy function](https://developer.mozilla.org/en-US/docs/Web/API/Node/isSameNode):
```suggestion
            const elementId = Array.from(options.componentRefs).find(([_, element]) => 
              getTemplateRefElement(unref(element))? === entry.target
            )?.[0];
```